File: programme/random-prime-number/random-prime-number.cs
// Write a programme to generate a random prime number in range n1-n2
using System;
using System.Collections.Generic;

class RandomPrimeNumber
{
    public static void Main(string[] args)
    {
        int
            a,
            b,
            i,
            j,
            flag;

        Console.Write("Enter lower bound of the interval : ");

        a = int.Parse(Console.ReadLine());

        Console.Write("\nEnter upper bound of the interval : ");

        b = int.Parse(Console.ReadLine());

        Console
            .Write("\nA random prime number between {0} and {1} is : ", a, b);
        List<int> numbers = new List<int>();

        for (i = a; i <= b; i++)
        {
            if (i == 1 || i == 0) continue;

            flag = 1;

            for (j = 2; j <= i / 2; ++j)
            {
                if (i % j == 0)
                {
                    flag = 0;
                    break;
                }
            }

            if (flag == 1) numbers.Add(i);
        }
        Random rnd = new Random();
        int r = rnd.Next(numbers.Count);
        Console.WriteLine(numbers[r]);
    }
}

File: programme/random-prime-number/RandomPrimeNumber.java
// Write a programme to generate a random prime number in range n1-n2

import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.util.Random;

public class RandomPrimeNumber{

    public static void main(String[] args){

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the lower bound of the interval: ");
        int lower = scanner.nextInt();

        System.out.print("Enter Upper bound of interval: ");
        int upper = scanner.nextInt();

        if(lower >= upper){
            System.out.println("The lower bound needs to be smaller than the upper bound");
        }
        else{
            System.out.printf("The random prime number is: %d", findPrimeNumber(lower, upper));
        }
    }

    private static int findPrimeNumber(int lower, int upper) {
        List<Integer> integerList = new ArrayList<>();
        Random random = new Random();
        for(int i = lower; i < upper; i++){
            if(isPrime(i)){
                integerList.add(i);
            }
        }
        int listIndex = random.nextInt(integerList.size());
        return integerList.get(listIndex);

    }

    private static boolean isPrime(int n) {
        if (n < 2)
            return false;

        for (int i = 2; i < n; i++)
        {
            if (n % i == 0)
                return false;
        }
        return true;
    }
}
